accepted/2025/cli-acquisition-tool.md

```diff
@@ -0,0 +1,224 @@
+* `DOTNET_ROOT` will be set to point to the `dnup` user-local root,
+* `PATH` will have the `dnup` user-local root appended to it,
+* on Windows `dnup-init` will request to elevate to set a Windows Registry key that will prevent global .NET installers from setting the global .NET Install location on the System `PATH` (which would always overwrite the user `PATH`).
+  * this behavior _may_ be optional - we may choose to only run it by default if the user has a global .NET install already
+
+This configuration will be necessity be shell-specific, so the tool will need to detect the user's shell and modify the appropriate profile file (e.g. `.bashrc`, `.zshrc`, etc.) to set the `DOTNET_ROOT` and `PATH` variables.
+We have some prior art in the `dotnet` CLI codebase for detecting the shell that is in use, which can be reused here.
+The different shells (and to some extent different OS platforms) have different ways of distributing such changes, so the tool will need to know about those to be a bit opinionated about how it makes these changes.
+
+This shell-specific configuration will need to be independently run via a command, because the user may choose a different shell after the initial setup.
+
+### Install locations
+
+These installs will be located at a default _user-local hive_ instead of a globally-reachable system hive.
+This will allow users to manage new tooling versions without needing to elevate.
+The tool will support being configured to install to other locations via configuration file, so that simple use of the tool remains consistent.
+
+### Install provenance
+
+The data about installs will be sourced from the release manifests that are published on a regular cadence by the .NET Releases team.
+These manifests, and the release artifact data and signatures they contain, will enable the tool to confirm the provenance of any toolsets downloaded to the users' computer.
+
+### Update Management
+
+The tool will enable update checks via these same manifests.
+As highly-cacheable static files, they will be easy to check for updates on a regular cadence, and user interactions with the `dnup` tool (directly via a command like `dnup update --check` or similar, or implicitly via IDE or `dotnet` tooling calling such commands) will be managed on a default (but user definable) cadence such as `daily`, `weekly`, etc.
+
+### Self-updates
+
+`dnup` needs to be able to update itself as well, via a command similar to `dnup update --self`.
+This command will check documented, signed release manifests for `dnup` and download + replace the current `dnup` binary with the new one.
+Provenance and signature validation of the new binary will also happen to ensure the content is as expected.
+This update capability should be able to be triggered manually or on an automatic cadence (e.g. every execution, once a day, etc.)
+
+## Q & A
+
+### How will I get `dnup`?
+
+We intend to distribute `dnup` via a direct download initially.
```

Comment:
note: rustup has a shell script for non-windows here - we need to look into _why_: sh.rustup.rs

Reply:
Yeah - turns out we'd need the same, if for no other reason than a user would need to chmod the newly-downloaded binary to make it executable.

Reply:
Yep! Looking at it in more detail this all seems a) doable and b) necessary for a good Linux onboarding experience. The bulk of the 'hard' logic (thinking here things like signature validation, knowledge of side-by-side installations and which parts of the layout are versioned vs unversioned) would still be in the NAOT binary, but the detection of _which_ binary to initially download would need to happen in the script.